### PR TITLE
Improve devtools experience when navigating

### LINKED ISSUE #15425: devtools can't handle new pipelines
Refreshes now give a different pipeline, and when you navigate to a new page, you also get a new pipeline. Devtools can currently only track a single pipeline (and it can't show historical data), so we can't effectively track network activity with it.

From discussion with @jdm:
We need some sort of coordination from the constellation (maybe?) to tell the devtools that pipeline 1 is navigating to pipeline 2 and the devtools should report stuff about pipeline 2 now

### PR DESCRIPTION
**Pull request by jdm**
The primary motivation for this work was to fix #15425, and these changes make it possible to use the devtools to meaningfully inspect multiple pages when navigating between them. Navigating through session history is not yet supported.

These changes also include improvements to the dedicated worker support, which broke at some point. We now can observe console messages in workers.

### CODE REVIEW

**Comment by gterzian** [comment]
I would be happy to review this if @asajeffrey has other things to do, I started casually looking at this yesterday already...

In any case, I do have one suggestion: instead of using the `BrowsingContextId` as a kind of unique identifier that can survive navigation from one pipeline to another, how about we introduce a new `DevtoolsId` or similar in the `msg` crate, to be used for that purpose? 

I think that would make it easier to use in the context of the Service worker as well, since unlike the dedicated worker that one cannot easily be tied to a BC(which I guess is why this PR doesn't try to do this).

**Comment by jdm** [comment]
I would rather file that as a possible improvement, since service worker support is only hypothetical right now.

**Review by gterzian — APPROVED** [review]
Sounds good, then as far as I can tell it looks good to me.

File: components/devtools/actors/worker.rs [hunk 1]
```diff
@@ -19,11 +54,94 @@ impl Actor for WorkerActor {
...
+        Ok(match msg_type {
+            "attach" => {
+                let msg = AttachedReply {
+                    from: self.name(),
+                    type_: "attached".to_owned(),
+                    url: self.url.as_str().to_owned(),
+                };
+                self.streams.borrow_mut().push(stream.try_clone().unwrap());
+                stream.write_json_packet(&msg);
+                // FIXME: fix messages to not requir forging a pipeline for worker messages
```
Comment (gterzian) [inline components/devtools/actors/worker.rs:71]: nit: typo

**Comment by jdm** [comment]
@bors-servo r=gterzian
Thanks for reviewing!

**Comment by jdm** [comment]
@bors-servo retry
* #25513

**Pull request merged**